### ISSUE TITLE
Free memory gives wrong result on linux

### ISSUE THREAD

**Issue by b-spruck** [epics-modules/iocStats#24]
The free memory returned is not correct on linux. Problem: file system cache and buffers are counted as "free", but the system might not be able to free them on demand. Example: Any file residing in /dev/shm is counted as free memory.

reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773

In osdMemUsage.c, summing of of items shoudl be replaced by the value of MemAvailable

**Comment by ralphlange** [comment]
But only if the system's kernel is new enough to provide it.

**Comment by b-spruck** [comment]
sure. is there a reason to stick to old kernels?

**Comment by ralphlange** [comment]
Many installation use RHEL etc. which tend to update conservatively.
In terms of platform compatibility, EPICS code usually has to support 10-15 years of OS/compiler history.

**Comment by b-spruck** [comment]
let me rephrase it, is there a reason NOT to support new kernel feature?

**Comment by b-spruck** [comment]
else if (strcmp(title, "MemFree:") == 0 ||
                       strcmp(title, "Buffers:") == 0 ||
                       strcmp(title, "Cached:") == 0) {
                memfree += value * 1024;
                found++;
}else if (strcmp(title, "MemAvailable:") == 0) {
                memfree = value * 1024;
                found=99; // sorry depends on order ...

**Comment by ralphlange** [comment]
I can't avoid not to think of reasons not to support new kernel features.

Yeah, something along that is obviously fine.
